storage/innobase/log/log0recv.cc

```diff
@@ -1331,17 +1331,26 @@ dberr_t recv_apply_hashed_log_recs(log_t &log, bool allow_ibuf) {
 
     mutex_exit(&recv_sys->mutex);
 
+    /* Stop the recv_writer thread from issuing any LRU
+    flush batches. */
+    mutex_enter(&recv_sys->writer_mutex);
+
```

Comment:
Please check about recv_writer_thread() in previous 8.0 versions. 

I think we made a Merge mistake in 8.0.30. We never rely on recv_writer_thread() to do LRU list flushing.
From 8.0.30, we brought upstream logic as is.

During recovery too, our LRU threads will do the flushing.

Please revert the entire recv_writer_thread logic and this change too and test.

If it still fails, the fix should be something else.